chain/client/src/stateless_validation/chunk_validator/mod.rs

```diff
@@ -158,17 +211,20 @@ impl ChunkValidator {
         }
 
         let runtime_adapter = self.runtime_adapter.clone();
+        let cache = self.validation_result_cache.clone();
         self.validation_spawner.spawn("stateless_validation", move || {
             // processing_done_tracker must survive until the processing is finished.
-            let _processing_done_tracker_capture = processing_done_tracker;
+            let _processing_done_tracker_capture: Option<ProcessingDoneTracker> =
+                processing_done_tracker;
 
             match validate_chunk_state_witness(
                 state_witness,
                 pre_validation_result,
                 epoch_manager.as_ref(),
                 runtime_adapter.as_ref(),
             ) {
-                Ok(()) => {
+                Ok(validation_result) => {
+                    cache.put(prev_chunk_hash, validation_result);
```

Comment:
Note: we could potentially save validation result to disk and merge the code paths above into one. I decided not to do that for a few reasons:
* Saving the result to disk will increase the disk footprint for chunk validators and we don't really need them most of the time
* Overloading "chunk extra" could make the code difficult to understand and debug. Keeping the code paths for chunk producer and chunk validator separate makes the code less elegant but easier to reason with.